secant/Util/RecoveryPhraseStorage.swift

```diff
@@ -0,0 +1,232 @@
+//
+//  RecoveryPhraseStorage.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 03/10/2022.
+//
+
+import Foundation
+
+/// Zcash implementation of the keychain that is not universal but designed to deliver functionality needed by the wallet itself.
+/// All the APIs should be thread safe according to official doc:
+/// https://developer.apple.com/documentation/security/certificate_key_and_trust_services/working_with_concurrency?language=objc
+/// For seamless usage the interface is designed to deal with simple types:
+// swiftlint:disable convenience_type
+final class RecoveryPhraseStorage {
+    enum Constants {
+        static let zcashBirthday = "zcashBirthday"
+        static let zcashRecoveryPhrase = "zcashRecoveryPhrase"
+    }
+
+    enum KeychainError: Error, Equatable {
+        case decoding
+        case duplicate
+        case encoding
+        case noDataFound
+        case unknown(OSStatus)
+    }
+
+    static let `default`:RecoveryPhraseStorage = RecoveryPhraseStorage()
+    
+    // MARK: - Query Helpers
+
+    static func baseQuery(forAccount account: String = "", andKey forKey: String) -> [String: Any] {
+        let query:[ String: AnyObject ] = [
+            /// Uniquely identify this keychain accessor
+            kSecAttrService as String: forKey as AnyObject,
+            kSecAttrAccount as String: account as AnyObject,
+            kSecClass as String: kSecClassGenericPassword,
+            /// The data in the keychain item can be accessed only while the device is unlocked by the user.
+            /// This is recommended for items that need to be accessible only while the application is in the foreground.
+            /// Items with this attribute do not migrate to a new device.
+            /// Thus, after restoring from a backup of a different device, these items will not be present.
+            kSecAttrAccessible as String: kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+        ]
+        
+        return query
+    }
+    
+    static func restoreQuery(forAccount account: String = "", andKey forKey: String) -> [String: Any] {
+        var query = baseQuery(forAccount: account, andKey: forKey)
+        query[kSecMatchLimit as String] = kSecMatchLimitOne
+        query[kSecReturnData as String] = kCFBooleanTrue
+        query[kSecReturnRef as String] = kCFBooleanFalse
+        query[kSecReturnPersistentRef as String] = kCFBooleanFalse
+        query[kSecReturnAttributes as String] = kCFBooleanFalse
+        
+        return query
+    }
+}
+
+private extension RecoveryPhraseStorage {
+    // MARK: - Recovery Phrase Interface
```

Comment:
```suggestion
}

    // MARK: - Recovery Phrase Interface

private extension RecoveryPhraseStorage {
```

This doesn't look like an interface though, is that the right term?

Reply:
Maybe ?
// MARK: - Recovery Phrase Public Interface